data/config-schema/src/main/resources/net/corda/schema/configuration/stateManager/1.0/stateManager-db-properties.json

```diff
@@ -1,101 +1,125 @@
 {
   "$schema": "https://json-schema.org/draft/2019-09/schema",
   "$id": "https://corda.r3.com/net/corda/schema/configuration/stateManager/1.0/stateManager-db-properties.json",
-  "title": "State Manager Database Configuration Schema",
-  "description": "Configuration schema for the database section. Note that this configuration cannot be updated dynamically through the REST endpoint.",
+  "title": "State Manager configuration schema for database implementation",
+  "description": "Configuration schema for State Manager's database implementation. Note that this configuration cannot be updated dynamically through the REST endpoint.",
   "type": "object",
-  "properties": {
-    "database": {
-      "description": "Database settings.",
-      "type": "object",
-      "default": {},
-      "properties": {
-        "user": {
-          "description": "The database user.",
-          "type": "string"
-        },
-        "pass": {
-          "description": "The database password.",
-          "type": "string"
-        },
-        "jdbc": {
-          "description": "JDBC settings.",
-          "type": "object",
-          "default": {},
-          "properties": {
-            "url": {
-              "description": "The JDBC URL.",
-              "type": "string",
-              "default": "jdbc:postgresql://state-manager-db:5432/state_manager"
-            },
-            "driver": {
-              "description": "The JDBC driver.",
-              "type": "string",
-              "default": "org.postgresql.Driver"
+  "additionalProperties": {
+    "type": "object",
+    "properties": {
+      "type": {
+        "description": "The type of state manager.",
+        "type": [
+          "string",
+          "null"
+        ]
+      },
```

Comment:
This is already part of `corda.stateManager.json` and doesn't belong here as the `type` is a higher level property, right?

Reply:
I believe there's no reason why some state managers couldn't be DATABASE and some REDIS

Reply:
Agreed, but the `type` should be at the `stateManager` higher level, not `stateManager.databaseProperties` (or whatever), right?. The "general" state manager configuration is [here](https://github.com/corda/corda-api/blob/release/os/5.2/data/config-schema/src/main/resources/net/corda/schema/configuration/stateManager/1.0/corda.stateManager.json) already and that schema has the `type` property. When that property is set as `DATABASE`, we apply and validate this schema (`stateManager-db-properties.json`).

Reply:
But yes you're right the current state isn't correct as we'd need to pull this allOf - if into `stateManager-db-properties`

Reply:
This is the "config" schema that would hold state-manager config in config-read-service component, however, we actually never put this config onto kafka or make it editable. It might be a good opportunity to delete this config file (and corda.stateManager.json) and just have the corda.boot.json

Reply:
Yep, I believe the parent `json` works just fine as it is (and that will also allow you to remove the `additionalProperties` from my [other comment](https://github.com/corda/corda-api/pull/1452#discussion_r1464919035)).